cli/config/src/lib.rs

```diff
@@ -38,8 +39,23 @@ impl Config {
             return Ok(Some(xdg_path));
         }
 
-        let legacy_path = dirs::home_dir()
-            .ok_or_else(|| anyhow!("Cannot determine home directory"))?
+        if cfg!(target_os = "macos") {
+            let legacy_apple_path = etcetera::base_strategy::Apple::new()?
+                .data_dir() // `$HOME/Library/Application Support/`
+                .join("tree-sitter")
+                .join("config.json");
+            if legacy_apple_path.is_file() {
+                fs::copy(&legacy_apple_path, &xdg_path)?;
+                fs::remove_file(&legacy_apple_path)?;
```

Comment:
A [`rename`](https://linux.die.net/man/2/rename) would be faster than a [`copy`](https://man7.org/linux/man-pages/man2/copy_file_range.2.html) + [`unlink`](https://linux.die.net/man/2/unlink)

```suggestion
                fs::rename(&legacy_apple_path, &xdg_path)?;
```